src/lib/file-util.mjs

```diff
@@ -27,3 +29,136 @@ export function dirIsWriteable(path) {
 
   return true;
 }
+
+/**
+ * Checks if a file exists at the given path.
+ *
+ * @param {string} path - The path to the file.
+ * @returns {boolean} - Returns true if the file exists, otherwise false.
+ */
+function fileExists(path) {
+  try {
+    fs.readFileSync(path);
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+/**
+ * Class representing credentials management.
+ */
+export class Credentials {
+  /**
+   * Creates an instance of Credentials.
+   *
+   * @param {string} [filename=""] - The name of the credentials file.
+   */
+  constructor(filename = "") {
+    this.logger = container.resolve("logger");
+    this.exit = container.resolve("exit");
+    this.filename = filename;
+    this.credsDir = `${os.homedir()}/.fauna/credentials`;
+    if (!dirExists(this.credsDir)) {
+      fs.mkdirSync(this.credsDir, { recursive: true });
+    }
+    this.filepath = `${this.credsDir}/${this.filename}`;
+    if (!fileExists(this.filepath)) {
+      fs.writeFileSync(this.filepath, "{}");
+    }
+  }
+
+  /**
+   * Retrieves the value associated with the given key from the credentials file.
+   *
+   * @param {string} [key] - The key to retrieve the value for.
+   * @returns {any} - The value associated with the key, or the entire parsed content if no key is provided.
+   */
+  get(key) {
+    try {
+      // Open file for reading and writing without truncating
+      const fileContent = fs
+        .readFileSync(this.filepath, { flag: "r+" })
+        .toString();
+      if (!isJSON(fileContent)) {
+        this.logger.stderr(
+          "Credentials file contains invalid formatting: ",
+          this.filepath
+        );
+        this.exit(1);
```

Comment:
prefer to throw an error over logging & exiting (and note that the error constructor only takes one string), e.g.:
```
throw new Error(`Credentials file contains invalid formatting: ${this.filepath}`)
```